### PR TITLE
First of a set of patches around consul changes.

### PR DESCRIPTION
**Pull request by galthaus**
Update recipe to be runnable multiple times - ui install badness
Add variables to the system for ACL and security management.
Enable the consul agent to leave/join to hand bind_addr changes.

### CODE REVIEW

File: chef/cookbooks/consul/recipes/start-service.rb [hunk 1]
```diff
@@ -90,6 +92,19 @@
   end
 end
 
+# Check if we are running and our bind address changed.
+bind_addr=%x{consul members | grep `hostname` | awk '{ print $2 }'}
+if bind_addr && bind_addr != ""
+  if bind_addr =~ /^#{node[:consul][:bind_addr]}/
```
Comment (VictorLowther) [inline chef/cookbooks/consul/recipes/start-service.rb]: Why a regex here, instead of comparing the IP addresses involved?

Comment (galthaus) [inline chef/cookbooks/consul/recipes/start-service.rb]: well - lazy - bind addr is ip:port and bind addr is just IP.  just wanted to match first, but I think I'll change it.  It doesn't work in all cases.  Time to not be lazy.

Comment (galthaus) [inline chef/cookbooks/consul/recipes/start-service.rb]: I'll fix it.

**Comment by galthaus** [comment]
Updated code.

**Comment by sjensen** [comment]
acl test from counsul

**Pull request merged**